src/proto/message_translation/power_frame_msg.hpp

```diff
@@ -0,0 +1,218 @@
+    {
+        fields = TbotsProto_PowerControl_fields;
+        size   = TbotsProto_PowerControl_size;
+    }
+    else if (std::is_same<T, TbotsProto_PowerStatus>::value)
+    {
+        fields = TbotsProto_PowerStatus_fields;
+        size   = TbotsProto_PowerStatus_size;
+    }
+    else
+    {
+        throw std::runtime_error("Unexpected type as argument");
+    }
+
+    std::vector<uint8_t> buffer(size);
+    pb_ostream_t stream =
+        pb_ostream_from_buffer(static_cast<uint8_t*>(buffer.data()), buffer.size());
+    if (!pb_encode(&stream, fields, &data))
+    {
+        throw std::runtime_error(
+            "Failed to encode PowerFrame msg to vector when converting nanopb to vector");
+    }
+    return buffer;
+}
+
+/**
+ * Helper function to set the type and value of a power frame's power_msg
+ *
+ * @param frame frame to set the power_msg of
+ * @param control/status power_msg to set
+ */
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerControl& control)
+{
+    frame.which_power_msg         = TbotsProto_PowerFrame_power_control_tag;
+    frame.power_msg.power_control = control;
+}
+
+void setPowerMsg(TbotsProto_PowerFrame& frame, const TbotsProto_PowerStatus& status)
+{
+    frame.which_power_msg        = TbotsProto_PowerFrame_power_status_tag;
+    frame.power_msg.power_status = status;
+}
+
+/**
+ * Creates a nanopb power status msg with provided fields
+ *
+ * @return a nanopb power status msg with provided fields
+ */
+TbotsProto_PowerStatus createNanoPbPowerStatus(float battery_voltage,
+                                               float capacitor_voltage,
+                                               float current_draw,
+                                               float high_voltage_measurement_volts,
+                                               float geneva_angle_deg,
+                                               bool breakbeam_tripped, bool flyback_fault)
+{
+    TbotsProto_PowerStatus status         = TbotsProto_PowerStatus_init_default;
+    status.battery_voltage                = battery_voltage;
+    status.capacitor_voltage              = capacitor_voltage;
+    status.current_draw                   = current_draw;
+    status.high_voltage_measurement_volts = high_voltage_measurement_volts;
+    status.geneva_angle_deg               = geneva_angle_deg;
+    status.breakbream_tripped             = breakbeam_tripped;
+    status.flyback_fault                  = flyback_fault;
+    return status;
+}
+
+// These functions are only used in power service so they are excluded from platformio
+// Protobuf needs additional support to run on an esp32
+#ifndef PLATFORMIO_BUILD
+/**
+ * Converts a google protobuf power control msg to its nanopb representation
+ *
+ * @param google_control protobuf message to convert
+ * @return a nanopb power control msg matching provided protobuf
+ */
+TbotsProto_PowerControl createNanoPbPowerControl(
```

Comment:
why do we need nanopb in C++

Reply:
This is needed so that we can convert the google proto powerControl inside the DirectControlPrimitive to nanopb powerControl that we can send to the esp32 to use.